scripts/core/editor3/reducers/tests/suggestions_insert.spec.jsx

```diff
@@ -0,0 +1,193 @@
+import {EditorState, ContentState} from 'draft-js';
+import * as Highlights from '../../helpers/highlights';
+import reducer from '../suggestions';
+
+describe('editor3.reducers.suggestion.CREATE_ADD_SUGGESTION', () => {
+    it('CREATE_ADD_SUGGESTION - insert text on empty content', () => {
+        const initialEditorState = EditorState.createEmpty();
+        const date = new Date();
+
+        const result = reducer({
+            editorState: initialEditorState,
+            suggestingMode: true,
+            onChangeValue: () => ({}),
+        }, {
+            type: 'CREATE_ADD_SUGGESTION',
+            payload: {
+                text: 'test',
+                data: {
+                    date: date,
+                    author: 'author_id',
+                },
+            },
+        });
+
+        const {editorState} = result;
+        const block = editorState.getCurrentContent().getFirstBlock();
+
+        expect(editorState.getCurrentContent().getPlainText()).toEqual('test');
+
+        for (let i = 0; i < 4; i++) {
+            expect(block.getInlineStyleAt(i).toJS()).toEqual(['ADD_SUGGESTION-1']);
+        }
+
+        expect(Highlights.getHighlightData(editorState, 'ADD_SUGGESTION-1')).toEqual({
+            date: date,
+            author: 'author_id',
+            type: 'ADD_SUGGESTION',
+        });
+    });
+
+    it('CREATE_ADD_SUGGESTION - insert text on middle of existing text', () => {
+        const content = ContentState.createFromText('initial');
+        const initialEditorState = EditorState.createWithContent(content);
+        const selection = initialEditorState.getSelection().merge({
+            anchorOffset: 2,
+            focusOffset: 2,
+        });
+        const date = new Date();
+
+        const result = reducer({
+            editorState: EditorState.acceptSelection(initialEditorState, selection),
+            suggestingMode: true,
+            onChangeValue: () => ({}),
+        }, {
+            type: 'CREATE_ADD_SUGGESTION',
+            payload: {
+                text: 'test',
+                data: {
+                    date: date,
+                    author: 'author_id',
+                },
+            },
+        });
+
+        const {editorState} = result;
+        const block = editorState.getCurrentContent().getFirstBlock();
+
+        expect(editorState.getCurrentContent().getPlainText()).toEqual('intestitial');
+
+        for (let i = 0; i < 11; i++) {
+            if (i > 1 && i < 6) {
+                expect(block.getInlineStyleAt(i).toJS()).toEqual(['ADD_SUGGESTION-1']);
```

Comment:
With getHighlightTypeFromStyleName you lost the identity of the suggestion and to compensate this you will need other tricks. 
I think that by using the current generated codes it is very easy to track the suggestions over the test code.
I don't see any reason to change the way we generate the suggestion ids, and even if we change it we can mock on tests the ids generation function with the current one.

Reply:
Take a look at the last two test. Here you need to know if there is a new suggestion is added or the old one is extended. Those are really simple tests, there will be more complicated later and I think to use those ids in tests is just fine.
Unit tests are mock friendly, so it is really a solution not a workaround.
It is not a good idea to over-complicate the tests just to avoid a possible but improbable problem that anyway has a simple solution.

Reply:
I agree that you do need to know the id. I think a better solution would be to get `CREATE_ADD_SUGGESTION` function to return the id of the created suggestion, but that's not how reducers work.

I guess the conclusion is that you can not use reducers if you want to hide implementation details of a functionality.